vodesfunc/descale.py

```diff
@@ -265,13 +285,16 @@ def get_upscaled(self, clip: vs.VideoNode, chroma: vs.VideoNode | None = None) -
                     mask = mask_fun.edgemask(self.doubled, self.line_mask[2], self.line_mask[3], planes=0)
                 self.line_mask = Scaler.ensure_obj(self.line_mask[1]).scale(mask, clip.width, clip.height)
 
-            if self.border_handling:
-                self._add_border_mask()
-
-            self.upscale = y.std.MaskedMerge(self.upscale, self.line_mask)
+        if self.border_handling:
+            self._add_border_mask()
 
-        if self.credit_mask != False or self.credit_mask_thr <= 0:
-            self.upscale = self.upscale.std.MaskedMerge(y, self.credit_mask)
+        if isinstance(self.credit_mask, vs.VideoNode) and isinstance(self.line_mask, vs.VideoNode):
```

Comment:
If we're going for efficiency anyway, you can assign these to a variable with the walrus operator here. May save two ifinstance calls.